### PR TITLE
Extend await docs to warn about recycling

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**

### CODE REVIEW

File: src/ocean/task/Scheduler.d [hunk 1]
```diff
@@ -442,6 +442,13 @@ final class Scheduler
 
         This method must not be called outside of a task.
 
+        Because of how termination hooks are implemented, by the time `await`
+        returns, task object is not yet completely recycled - it will only
+        happen during next content switch. Caller of `await` must either ensure
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/task/Scheduler.d]: `s/content switch/context switch/` ?

File: src/ocean/task/Scheduler.d [hunk 2]
```diff
@@ -442,6 +442,13 @@ final class Scheduler
...
+        task object live long enough for that or call
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/task/Scheduler.d]: `ensure that task object live long enough`? Not sure about this one, though. @gavin-norman-sociomantic ?

**Comment by mihails-strasuns-sociomantic** [comment]
Threw in some articles too.

**Comment by nemanja-boric-sociomantic** [comment]
Yeah, I can't help you there ;-)

**Review by gavin-norman-sociomantic — APPROVED** [review]
Nice sprinkling of articles :)

**Pull request merged**